### PR TITLE
Fearture/reusable components 2

### PR DESCRIPTION
**Pull request by Naiby**

### CODE REVIEW

**Review by AlefrankM — CHANGES_REQUESTED** [review]
Refactorize

File: src/components/common/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    | 'password'
+    | 'search'
+    | 'submit'
+    | 'text'
+    | 'tel'
+    | 'url'
+}
+
+export function Input({ hover, placeholder, type }: Props) {
+  let Icon
```
Comment (juliswer) [inline src/components/common/Input.tsx:35]: why are you using a let variable instead of a next state?

**Review by juliswer — APPROVED** [review]
Good job! Please check my comment

File: src/components/common/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  switch (type) {
```
Comment (AlefrankM) [inline src/components/common/Input.tsx:36]: We can use a dinamyc object instead of switch

File: src/components/common/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  AiFillFile,
+  AiTwotoneCalendar,
+  AiTwotoneEdit,
+  AiTwotoneLock,
+  AiTwotoneMail,
+} from 'react-icons/ai'
+
+type Props = {
+  hover:
+    | 'primary'
```
Comment (AlefrankM) [inline src/components/common/Input.tsx:11]: Can you create an enum or use an existing?

File: src/components/common/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    | 'email'
+    | 'date'
+    | 'file'
+    | 'hidden'
+    | 'image'
+    | 'number'
+    | 'password'
+    | 'search'
+    | 'submit'
+    | 'text'
```
Comment (AlefrankM) [inline src/components/common/Input.tsx:29]: default is text

File: src/components/common/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  hover:
+    | 'primary'
+    | 'secondary'
+    | 'accent'
+    | 'info'
+    | 'success'
+    | 'warning'
+    | 'error'
+  placeholder: string
+  type:
```
Comment (AlefrankM) [inline src/components/common/Input.tsx:19]: Convert to Enum

File: src/components/common/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  return (
+    <div className="relative m-1 justify-center ">
+      {type !== 'date' && (
+        <div className="absolute inset-y-0 left-0 w-16 items-center pl-3 pt-3 pointer-events-none ">
+          {Icon}
+        </div>
+      )}
+      <input
+        className={`input ${
+          type === 'file' ? 'file-input-ghost' : ''
```
Comment (AlefrankM) [inline src/components/common/Input.tsx:62]: You have to avoid this magic strings

File: src/layout/Header.tsx [hunk 7]
```diff
@@ -1,9 +1,13 @@
...
-export function Header() {
+type Props = {
+  background: 'bg-indigo-900' | 'bg-gray-100'
+}
+
+export function Header({ background }: Props) {
   return (
     <header>
-      <Navbar />
+      <Navbar background={background} />
```
Comment (AlefrankM) [inline src/layout/Header.tsx:10]: What is this? A component just to call another component?

File: src/styles/tailwind.css [hunk 8]
```diff
@@ -27,6 +27,9 @@
   .bg-bluedark-100 {
     background-color: rgb(var(--color-secondary-dark));
   }
+  .bg-bluelight-300 {
```
Comment (AlefrankM) [inline src/styles/tailwind.css:30]: Please, you can use tailwind.config for these custom colors

**Review by AlefrankM — CHANGES_REQUESTED** [review]
Please

**Review by AlefrankM — DISMISSED** [review]

**Review by AlefrankM — APPROVED** [review]

**Pull request merged**